playwright-e2e/dsm/pages/miscellaneous-pages/user-and-permissions-page.ts

```diff
@@ -0,0 +1,307 @@
+    return studyAdmin.locator(`/following-sibling::mat-panel-description//button[@mattooltip='Delete User']`);
+  }
+
+  public async getPermissionsSection(studyAdmin: Locator): Promise<Array<Locator>> {
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//section[contains(@class, 'permissions')]//app-permission-checkbox`).all();
+  }
+
+  public getSaveButton(studyAdmin: Locator): Locator {
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//button[contains(., 'Save')]`);
+  }
+
+  public getDiscardChangesButton(studyAdmin: Locator) {
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//button[contains(., 'Discard changes')]`);
+  }
+
+  /* Assertions */
+  public async assertPageTitle(): Promise<void> {
+    await expect(this.page.getByRole('heading', { name: `${this.PAGE_TITLE}` }),
+      `ERROR - ${this.PAGE_TITLE}'s page title is not displayed`).toBeVisible();
+  }
+
+  public async assertAddUserButtonDisplayed(): Promise<void> {
+    const button = this.getAddUserButton();
+    await expect(button).toBeVisible();
+  }
+
+  public async assertStudyAdminInfo(email: string, name: string, phone?: string): Promise<void> {
+    //Get the relevant study admin's section
+    const studyAdmin = this.getStudyAdmin(email);
+
+    //Verify the email is as expected
+    const studyAdminEmail = await studyAdmin.innerText();
+    expect(studyAdminEmail).toBe(email);
+
+    //Verify the name is as expected
+    const studyAdminName = await this.getStudyAdminName(name);
+    expect(studyAdminName).toBe(name);
+
+    if (phone) {
+      //If phone information is provided, verify that it is shown in the page as expected
+      const studyAdminPhoneNumber = await this.getStudyAdminPhoneNumber(phone);
+      expect(studyAdminPhoneNumber).toBe(phone);
+    }
+  }
+
+  public async assertAllPossibleStudyPermissionsAreVisible(study: StudyEnum, email: string): Promise<void> {
+    //Verify that all possible permissions for the study are available for the study admin to select
+    const studyAdmin = this.getStudyAdmin(email);
+    const allPossiblePermissions = await this.getPermissionsSection(studyAdmin);
+    const amountOfPermissions = allPossiblePermissions.length;
+    console.log(`Amount of permissions: ${amountOfPermissions}`);
+
+    for (let index = 0; index < amountOfPermissions; index++) {
+      const permissionCheckbox = allPossiblePermissions[index];
+      const permission = permissionCheckbox.locator(`//span[@class='ng-star-inserted']`);
+      const currentPermission = (await permission.innerText()).trim();
+      //console.log(`Current permission: ${currentPermission}`);
+      const currentPermissionCastedAsEnum = currentPermission as UserPermission;
+      const verifiedPermission = Object.values(this.currentStudyPermissions).indexOf(currentPermissionCastedAsEnum) >= 0;
+      if (!verifiedPermission) {
+        throw new Error(`The ${currentPermission} permission is not a valid permission for ${study}`);
```

Comment:
Nice!  Love to see such crisp error messages--it makes troubleshooting much easier.